libc/stdio_printf.c

```diff
@@ -0,0 +1,3322 @@
+	vtest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+	vsntest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+	vstest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+}
+
+
+TEST(stdio_printf_i, zi)
+{
+	int i;
+	char expect[256] = { 0 };
+	const char *format = "%zi %zi %zi";
+	const size_t values[] = { SSIZE_MAX, SSIZE_MAX / 2, 0 };
+
+	for (i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		if (i > 0) {
+			strcat(expect, " ");
+		}
+		strcat(expect, signed_to_string(values[i], 10));
+	}
+
+	dtest(format, values[0] COMMA values[1] COMMA values[2], expect);
+	ftest(format, values[0] COMMA values[1] COMMA values[2], expect);
+	stdtest(format, values[0] COMMA values[1] COMMA values[2], expect);
+	sntest(format, values[0] COMMA values[1] COMMA values[2], expect);
+	stest(format, values[0] COMMA values[1] COMMA values[2], expect);
+
+	vdtest(expect, format, values[0], values[1], values[2]);
+	vftest(expect, format, values[0], values[1], values[2]);
+	vtest(expect, format, values[0], values[1], values[2]);
+	vsntest(expect, format, values[0], values[1], values[2]);
+	vstest(expect, format, values[0], values[1], values[2]);
+}
+
+
+TEST(stdio_printf_i, ti)
+{
+	int i;
+	char expect[256] = { 0 };
+	const char *format = "%ti %ti %ti %ti %ti";
+	const ptrdiff_t values[] = { PTRDIFF_MAX, PTRDIFF_MAX / 2, 0, PTRDIFF_MIN / 2, PTRDIFF_MIN };
+
+	for (i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		if (i > 0) {
+			strcat(expect, " ");
+		}
+		strcat(expect, signed_to_string(values[i], 10));
+	}
+
+	dtest(format, values[0] COMMA values[1] COMMA values[2] COMMA values[3] COMMA values[4], expect);
+	ftest(format, values[0] COMMA values[1] COMMA values[2] COMMA values[3] COMMA values[4], expect);
+	stdtest(format, values[0] COMMA values[1] COMMA values[2] COMMA values[3] COMMA values[4], expect);
+	sntest(format, values[0] COMMA values[1] COMMA values[2] COMMA values[3] COMMA values[4], expect);
+	stest(format, values[0] COMMA values[1] COMMA values[2] COMMA values[3] COMMA values[4], expect);
+
+	vdtest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+	vftest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+	vtest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+	vsntest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+	vstest(expect, format, values[0], values[1], values[2], values[3], values[4]);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+TEST_SETUP(stdio_printf_o)
+{
+	output_file = fopen(PATH, "w+");
+}
+
+
+TEST_TEAR_DOWN(stdio_printf_o)
+{
+	fclose(output_file);
+	remove(PATH);
+}
+
+
+TEST(stdio_printf_o, o)
+{
+	int i;
+	char expect[256] = { 0 };
+	const char *format = "%o %o %o %o %o %o %o";
+	const unsigned int values[] = { UINT_MAX, UINT_MAX / 2, INT_MAX, INT_MAX / 2, 0, INT_MIN / 2, INT_MIN };
```

Comment:
Why ints are passed to unisgned int array? You may want to tests values out of bonds, but I think we can do it in separate cases, because as you can see you would need the another type.

Reply:
Applies to the other places like this one.